### ISSUE TITLE
Coverage only works when individual test files are run

### ISSUE THREAD

**Issue by OutOfFocus4** [denoland/deno#12751]
I'm working on a project using Deno, and I noticed while test my code that the `deno coverage` command was listing some lines of code as not covered even though they **had** to be covered for the tests to pass.

I managed to replicate the issue with the following files:

```
/tmp/example $ ls
example.ts  one_test.ts  two_test.ts
```

```ts
// example.ts
export function showExample(value1: unknown, value2: unknown) {
	if (typeof value2 !== 'boolean') {
		throw new TypeError(`'value2' should be a boolean`);
	}
	if (value2) {
		return String(value1);
	}
	return value1;
}

// one_test.ts
import { assertEquals } from "https://deno.land/std@0.114.0/testing/asserts.ts";

import { showExample } from './example.ts';

Deno.test({
	name: 'value2 is boolean test',
	fn() {
		assertEquals(showExample(1, true), '1');
	}
});

// two_test.ts
import * as asserts from "https://deno.land/std@0.114.0/testing/asserts.ts";

import { showExample } from './example.ts';

Deno.test({
	name: 'value2 is false test',
	fn() {
		const b = ['test'];
		asserts.assertEquals(showExample(b, false), b);
	}
});
```

I then run the tests and check the coverage:

```
/tmp/example $ mkdir .deno
/tmp/example $ DENO_DIR=./.deno deno test --coverage=cov
Download https://deno.land/std@0.114.0/testing/asserts.ts
Download https://deno.land/std@0.114.0/fmt/colors.ts
Download https://deno.land/std@0.114.0/testing/_diff.ts
Check file:///tmp/example/one_test.ts
Check file:///tmp/example/two_test.ts
running 1 test from file:///tmp/example/one_test.ts
test value2 is boolean test ... ok (11ms)
running 1 test from file:///tmp/example/two_test.ts
test value2 is false test ... ok (13ms)

test result: ok. 2 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out (143ms)

/tmp/example $ DENO_DIR=./.deno deno coverage cov
Check file:///tmp/example/example.ts
cover file:///tmp/example/example.ts ... 22.222% (2/9)
   2 | 	if (typeof value2 !== 'boolean') {
   3 | 		throw new TypeError(`'value2' should be a boolean`);
   4 | 	}
   5 | 	if (value2) {
   6 | 		return String(value1);
   7 | 	}
   8 | 	return value1;
```

The tests passed, but the code coverage shows `showExample` was never called.

If I just run the test in `one_test.ts` on its own, though...

```
/tmp/example $ rm -rf cov
/tmp/example $ DENO_DIR=./.deno deno test --coverage=cov one_test.ts 
Check file:///tmp/example/one_test.ts
Check file:///tmp/example/one_test.ts
running 1 test from file:///tmp/example/one_test.ts
test value2 is boolean test ... ok (12ms)

test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out (956ms)

/tmp/example $ DENO_DIR=./.deno deno coverage cov
cover file:///tmp/example/example.ts ... 44.444% (4/9)
   2 | 	if (typeof value2 !== 'boolean') {
   3 | 		throw new TypeError(`'value2' should be a boolean`);
   4 | 	}
-----|-----
   7 | 	}
   8 | 	return value1;
```

... then the code coverage is correct. I'm not familiar enough with Deno's internals to guess why this is occurring.

**Comment by zandaqo** [comment]
This is likely related to #11875 I too have full coverage when testing individual files, e.g.
```bash
deno test -A --coverage=cov .\tests\adjacency-list_test.ts
```
shows full coverage of `adjacency-list.ts`, whereas testing all files with `deno test -A --coverage=cov` shows missing coverage in the `adjacency-list.ts`.

**Comment by GJZwiers** [comment]
Looks similar to [a comment](https://github.com/denoland/deno/issues/10944#issuecomment-860262706) in #10944

**Comment by bartlomieju** [comment]
I confirmed this is the problem and most likely the same cause as #10944, should be fixed by #13334
